### PR TITLE
mu: use EmacsRequirement

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: Library/Formula/mu.rb [hunk 1]
```diff
@@ -32,7 +21,7 @@ class Mu < Formula
   depends_on "glib"
   depends_on "gmime"
   depends_on "xapian"
-  depends_on Emacs23Installed if build.with? "emacs"
+  depends_on :emacs => "23" if build.with? "emacs"
```
Comment (xu-cheng) [inline Library/Formula/mu.rb]: Should be `depends_on :emacs => ["23", :optional]`

Comment (dunn) [inline Library/Formula/mu.rb]: Ah, right.

File: Library/Formula/mu.rb [hunk 2]
```diff
@@ -32,7 +21,7 @@ class Mu < Formula
...
 
   env :std if build.with? "emacs"
```
Comment (xu-cheng) [inline Library/Formula/mu.rb]: Also drop this line.

File: Library/Formula/mu.rb [hunk 3]
```diff
@@ -32,7 +21,7 @@ class Mu < Formula
   depends_on "glib"
   depends_on "gmime"
   depends_on "xapian"
-  depends_on Emacs23Installed if build.with? "emacs"
+  depends_on :emacs => ["23", :optional]
 
   env :std if build.with? "emacs"
```
Comment (xu-cheng) [inline Library/Formula/mu.rb]: Can this be dropped?

Comment (dunn) [inline Library/Formula/mu.rb]: Doesn't seem necessary on Yosemite. I'll try removing it.

**Comment by xu-cheng** [comment]
Thanks @dunn

**Pull request closed without merging**